exchanges/gateio/gateio_ws_futures.go

```diff
@@ -103,41 +103,39 @@ func (g *Gateio) GenerateFuturesDefaultSubscriptions(settlement currency.Code) (
 		return nil, err
 	}
 
-	var subscriptions subscription.List
-	for i := range channelsToSubscribe {
-		switch {
-		case settlement.Equal(currency.USDT):
-			pairs, err = pairs.GetPairsByQuote(currency.USDT)
-			if err != nil {
-				return nil, err
-			}
-		case settlement.Equal(currency.BTC):
-			offset := 0
-			for x := range pairs {
-				if pairs[x].Quote.Equal(currency.USDT) {
-					continue // skip USDT pairs
-				}
-				pairs[offset] = pairs[x]
-				offset++
+	settlementPairs := make(currency.Pairs, 0, len(pairs))
+	switch {
+	case settlement.Equal(currency.USDT):
+		settlementPairs, err = pairs.GetPairsByQuote(currency.USDT)
+		if err != nil {
+			return nil, err
+		}
+	case settlement.Equal(currency.BTC):
+		for x := range pairs {
+			if pairs[x].Quote.Equal(currency.USDT) {
+				continue // skip USDT pairs
```

Comment:
🐛 🐻  ( Bug Bear )
This comment says what the code is doing. Which we can read.
Comments should (mostly) explain *why* something is being done.
I know you didn't add this in this PR, but if you know the answer, can you improve this comment?